frontend_tests/node_tests/compose.js

```diff
@@ -2,9 +2,22 @@ set_global('$', global.make_zjquery());
 set_global('i18n', global.stub_i18n);
 
 set_global('page_params', {
-    use_websockets: false,
+    use_websockets: true,
 });
 
+var socket_send_checked = false;
+set_global('Socket', function () {
+    this.send = function (request, success, error) {
```

Comment:
Let's use the testing pattern below (with socket.send instead of $.ajax).  This pattern gives you more control to explicitly call the success and error functions.  See `frontend_tests/node_tests/channel.js` for more context.

~~~ js
  9 function test_with_mock_ajax(test_params) {
 10     var ajax_called;
 11     var ajax_options;
 12 
 13     $.ajax = function (options) {
 14         $.ajax = undefined;
 15         ajax_called = true;
 16         ajax_options = options;
 17         var xhr = test_params.xhr || default_stub_xhr;
 18 
 19         options.simulate_success = function (data, text_status) {
 20             options.success(data, text_status, xhr);
 21         };
 22 
 23         options.simulate_error = function () {
 24             options.error(xhr);
 25         };
 26 
 27         return xhr;
 28     };
 29 
 30     test_params.run_code();
 31     assert(ajax_called);
 32     test_params.check_ajax_options(ajax_options);
 33 }